tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/c_api/AbstractTF_Buffer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ =======================================================================
+ */
+
+package org.tensorflow.c_api;
+
+import org.bytedeco.javacpp.*;
+import org.bytedeco.javacpp.annotation.Properties;
+
+import static org.tensorflow.c_api.global.tensorflow.*;
+
+@Properties(inherit = org.tensorflow.c_api.presets.tensorflow.class)
+public abstract class AbstractTF_Buffer extends Pointer {
+    protected static class DeleteDeallocator extends TF_Buffer implements Pointer.Deallocator {
+        DeleteDeallocator(TF_Buffer s) { super(s); }
+        @Override public void deallocate() { if (!isNull()) TF_DeleteBuffer(this); setNull(); }
+    }
+
+    public AbstractTF_Buffer(Pointer p) { super(p); }
+
+    /**
+     * Calls TF_NewBuffer(), and registers a deallocator.
+     * @return TF_Buffer created. Do not call TF_DeleteBuffer() on it.
+     */
+    public static TF_Buffer newBuffer() {
+        TF_Buffer b = TF_NewBuffer();
+        if (b != null) {
+            b.deallocator(new DeleteDeallocator(b));
+        }
+        return b;
+    }
+
+    /** Returns {@code newBufferFromString(new BytePointer(proto)). */
+    public static TF_Buffer newBufferFromString(byte[] proto) {
+        return newBufferFromString(new BytePointer(proto));
+    }
+
+    /**
+     * Calls TF_NewBufferFromString(), and registers a deallocator.
+     * @return TF_Buffer created. Do not call TF_DeleteBuffer() on it.
+     */
+    public static TF_Buffer newBufferFromString(Pointer proto) {
+        TF_Buffer b = TF_NewBufferFromString(proto, proto.limit());
+        if (b != null) {
+            b.deallocator(new DeleteDeallocator(b));
+        }
+        return b;
+    }
+
+    /**
+     * Calls the deallocator, if registered, otherwise has no effect.
+     */
+    public void delete() {
+        deallocate();
+    }
+}
```

Comment:
Is this class (and all other classes found in `src/main/java/org/tensorflow/c_api`) actually used by your PR? If not, I would prefer we wait for the second phase before adding them.

Reply:
Well, they are the base classes of `TF_Buffer`, etc. They are functional, and we can modify them later as well. What are you worried about?

Reply:
I just don't find any additional value for having classes in the code base that are actually not used, it creates unnecessary confusion and they can be easily reviewed and added later as part of the whole migration of the JNI code.

Reply:
But they are used. Why do you say that they are not?

Reply:
Oops, sorry I didn’t read carefully that last reply of yours, all good then.

So generated classes can extend from human-written ones? Just for my knowledge, where this magic is happening?

Reply:
Ok got it: https://github.com/saudet/tensorflow-java/blob/18ea0b7939442d4539a61c8c0a164bc15f9ead51/tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/c_api/presets/tensorflow.java#L118